src/librbd/internal.cc

```diff
@@ -3513,4 +3513,143 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
       ictx->perfcounter->inc(l_librbd_readahead_bytes, readahead_length);
     }
   }
+
+  // Consistency groups functions
+
+  int create_cg(librados::IoCtx& io_ctx, const char *cg_name)
+  {
+    ostringstream bid_ss;
+    uint32_t extra;
+    int remove_r;
+    string id, id_cg, header_oid;
+
+    CephContext *cct = (CephContext *)io_ctx.cct();
+
+    Rados rados(io_ctx);
+    uint64_t bid = rados.get_instance_id();
+
+    id_cg = util::id_cg_name(cg_name);
```

Comment:
No, you would still want the header.  Naming it ```cg_header.<name>``` introduces race conditions if you crash during a rename.  I don't think looking up the id in the directory is a big deal -- you'd have to look it up in the id object anyway.